storage/shard.go

```diff
@@ -990,31 +957,64 @@ func (s *dbShard) tryRetrieveWritableSeries(id ident.ID) (
 	return nil, opts, nil
 }
 
-func (s *dbShard) newShardEntry(id ident.ID, tags ident.TagIterator) (*dbShardEntry, error) {
-	clonedTags, err := s.cloneTags(tags)
-	if err != nil {
-		return nil, err
-	}
+func (s *dbShard) newShardEntry(
+	id ident.ID,
+	tags ident.TagIterator,
+) (*lookup.Entry, error) {
 	series := s.seriesPool.Get()
-	clonedID := s.identifierPool.Clone(id)
-	series.Reset(clonedID, clonedTags, s.seriesBlockRetriever,
-		s.seriesOnRetrieveBlock, s, s.seriesOpts)
-	uniqueIndex := s.increasingIndex.nextIndex()
-	return &dbShardEntry{series: series, index: uniqueIndex}, nil
-}
-
-func (s *dbShard) cloneTags(tags ident.TagIterator) (ident.Tags, error) {
+	// NB(r): As documented in storage/series.DatabaseSeries the series IDs
+	// are garbage collected, hence we cast the ID to a BytesID that can't be
+	// finalized.
+	// Since series are purged so infrequently the overhead of not releasing
+	// back an ID to a pool is amortized over a long period of time.
+	clonedID := ident.BytesID(append([]byte(nil), id.Bytes()...))
```

Comment:
any reason to not call NoFinalize() here?

Reply:
ident.BytesID is a []byte, can't finalize it - https://github.com/m3db/m3x/blob/master/ident/bytes_id.go#L27-L55

Might be worth a comment indicating that here.

Reply:
Or better yet, just make the NoFinalize call. It'll get compiled away anyway

Reply:
Sure thing.